aptos-move/e2e-move-tests/src/tests/object_code_deployment.rs

```diff
@@ -362,6 +373,37 @@ fn freeze_code_object_fail_when_not_owner() {
     context.assert_feature_flag_error(status, ENOT_PACKAGE_OWNER);
 }
 
+#[test]
+fn freeze_code_object_fail_when_having_mutable_dependency() {
+    let mut context = TestContext::new(None, None);
+    let acc = context.account.clone();
+
+    assert_success!(context.execute_object_code_action(
+        &acc,
+        "object_code_deployment.data/pack_upgrade_compat",
+        ObjectCodeAction::Deploy,
+    ));
+    let mut options = BuildOptions::default();
+    options
+        .named_addresses
+        .insert(MODULE_ADDRESS_NAME.to_string(), context.object_address);
+    let sequence_number = context.harness.sequence_number(acc.address());
+    context.object_address =
+        create_object_code_deployment_address(*acc.address(), sequence_number + 1);
+    options.named_addresses.insert(
+        MUT_DEPS_MODULE_ADDRESS_NAME.to_string(),
+        context.object_address,
+    );
+    assert_success!(context.execute_object_code_action_with_options(
+        &acc,
+        "object_code_deployment.data/pack_mutable_deps",
+        ObjectCodeAction::Deploy,
+        options
+    ));
+    let status = context.execute_object_code_action(&acc, "", ObjectCodeAction::Freeze);
```

Comment:
nit: Would be nice to have a comment mentioning freezing a package with mutable dependencies is not allowed.